Attendance: create Model component

erisjacey opened the pull request:
- Package seedu.address.model.attendance
- Class seedu.address.model.Attributes
- Class seedu.address.model.Class
- Class seedu.address.model.ClassList
- Class seedu.address.model.Participation
- Class seedu.address.model.Presence
- Class seedu.address.model.exceptions.DuplicateClassException
- All relevant test files for the above classes

Coming soon:
- Create Logic component
- Create Storage component

erisjacey committed:
Attendance: create Model component

- Package seedu.address.model.attendance
- Class seedu.address.model.Attributes
- Class seedu.address.model.Class
- Class seedu.address.model.ClassList
- Class seedu.address.model.Participation
- Class seedu.address.model.Presence
- Class seedu.address.model.exceptions.DuplicateClassException
- All relevant test files for the above classes

marcustw commented on src/main/java/seedu/address/model/attendance/Class.java:
```diff
@@ -0,0 +1,98 @@
+ * Represents a particular class at a particular date.
+ * Each Class contains a list of details for each student in ATAS for the TA to keep track of.
+ */
+public class Class implements Comparable<Class> {
```
Perhaps change the name of the class to Session

marcustw commented on src/main/java/seedu/address/model/attendance/Class.java:
```diff
@@ -0,0 +1,98 @@
+ */
+public class Class implements Comparable<Class> {
+
+    private final LocalDate date;
```
Will we support delete+?

erisjacey committed:
Add constant values

- Participation.java
- Presence.java

erisjacey committed:
Rename files

- Class.java => Session.java
- ClassList.java => SessionList.java
- ClassTest.java => SessionTest.java
- ClassListTest.java => SessionListTest.java
- DuplicateClassException.java => DuplicateSessionException.java

YuunoKun reviewed (approved):
LGTM
